share/functions/fish_delta.fish

```diff
@@ -0,0 +1,100 @@
+function fish_delta
+    argparse h/help f/no-functions c/no-completions d/no-diff n/new -- $argv
+
+    if set -q _flag_help
+        __fish_print_help fish_delta
+        return 0
+    end
+
+    type -q diff
+    or set -l _flag_d --no-diff
+
+    # TODO: Do we want to keep the vendor dirs in here?
+    set -l default_function_path $__fish_data_dir/functions
+    set -l default_complete_path $__fish_data_dir/completions
+
+    set -l user_function_path
+    set -l user_complete_path
+    
+    for dir in $fish_function_path
+        contains -- $dir $default_function_path
+        or set -a user_function_path $dir
+    end
+    for dir in $fish_complete_path
+        # We don't care about generated completions.
+        # They shouldn't be compared at all.
+        contains -- $dir $default_complete_path $__fish_user_data_dir/generated_completions
+        or set -a user_complete_path $dir
+    end
+
+    set -l vars
+    not set -ql _flag_f[1]
+    and set -a vars user_function_path default_function_path
+    not set -ql _flag_c[1]
+    and set -a vars user_complete_path default_complete_path
+
+    if not set -q vars[1]
+        echo No directories to check! >&2
+        return 2
+    end
+    
+    set -l have_diff 0
+
+    if isatty stdout
+        set -f colors "$(set_color normal)" "$(set_color brblue)" "$(set_color bryellow)" "$(set_color green)"
+        set -f diff_color --color=always
+        set -f pager less
+        set -q PAGER
+        and echo $PAGER | read -at pager
+        set -l lessopts isRF
+        if test (less --version | string match -r 'less (\d+)')[2] -lt 530 2>/dev/null
+            set lessopts "$lessopts"X
+        end
+
+        not set -qx LESS
+        and set -xf LESS $lessopts
```

Comment:
I wonder if theres a good way to extract the paging logic.
A function will probably not work because we need to pipe into it.. maybe we can create a function that outputs the pager command